### ISSUE TITLE
A model that computes its own loss must still be fed a dummy target on every TrainStep

### ISSUE THREAD

**Issue by Eric-Louchez** [Shorokoo/Shorokoo#331]
`TrainingRig` requires a loss graph with exactly two tensor inputs and one scalar output. When the
model already produces the loss — a language model returning the scalar mean cross-entropy directly,
so that `checkpoint.ToInferenceModel()` hands back something that computes a validation loss — the
loss module becomes a forwarder that ignores its second input, and the *caller* must nonetheless
construct and pass a value for that input on every step.

`training.md` documents this:

> **You still feed the ignored input.** `rig.TargetDef` is derived from the loss graph's second input
> whether or not the body reads it … every `TrainStep` still passes a target struct — omitting it
> fails the input-count check (`CR006`). Only its **dtype** must match … its shape and contents are
> ignored, so a zero-element placeholder is enough.

Confirmed on `main` (798953d), LinuxCPU backend:

```csharp
[Module]
public partial class SelfLossModel                       // returns its own scalar loss
{
    public static Scalar<float32> Inline(Tensor<float32> x) { ... }
}

[Module]
public partial class PassThroughLoss                     // the rig's loss slot
{
    public static Scalar<float32> Inline(Scalar<float32> predictions, Tensor<float32> targets)
        => predictions;
}
```

```
TargetDef fields: 1 (targets)
zero-element dummy target accepted, loss 0
omitting the target: ArgumentException: Missing data for field 'targets' (Parameter 'fields')
```

So the rig derives a mandatory input from a parameter the loss body demonstrably never reads.

## Why it is worth recording

The cost is one line and `TensorData(DType.Float32, [0L])` is free. The problem is what the call site
reads as:

```csharp
var noTargets = rig.TargetDef.FromOrderedData(TensorData(DType.Float32, [0L]));
...
ckpt = rig.TrainStep(ckpt, inputs, noTargets);
```

`TrainStep(ckpt, inputs, noTargets)` looks like "inputs and targets", but the *real* targets are
inside `inputs` — the model takes `(tokens, targets)` — and the argument in the target position is an
empty placeholder. A reader has to know about the arrangement to parse that line, and the variable
name is doing the work the API does not. The same placeholder appeared at three call sites.

The rig already knows the loss body ignores its second input (the body is inlined into the composed
training step), so the derivation could drop the target slot rather than requiring a value for it.

Related: #329 (why the loss was put inside the model in the first place).

**No migration work.** Shorokoo has no installed base to protect. Change `TrainStep`'s signature or
the `TargetDef` derivation outright — no compatibility shim and no version bump.
